test/support/xml_generator.ex

```diff
@@ -0,0 +1,245 @@
+defmodule Cadet.Test.XMLGenerator do
+  @moduledoc """
+  This module contains functions to produce sample XML codes in accordance to
+  the specification (xml_api.rst).
+
+  # TODO: Refactor using macros
+  """
+
+  alias Cadet.Assessments.{Assessment, Question}
+
+  import XmlBuilder
+
+  @spec generate_xml_for(%Assessment{}, [%Question{}], [{atom(), any()}]) :: String.t()
+  def generate_xml_for(
```

Comment:
Would prefer if this function were broken up into smaller chunks but I think it's not a priority for now. Can add a refactor issue instead.